cache/models/samples.js

```diff
@@ -250,6 +267,142 @@ function applyFiltersOnSampObjs(sampObjArray, opts) {
   return filteredSamples;
 }
 
+/**
+ * Create properties array having and fields and values need to be set to
+ * update/create sample.
+ * @param  {Object} qbObj - Query body object
+ * @param  {Object} sampObj - Sample object
+ * @param  {Object} aspectObj - Aspect object
+ * @return {Array} Properties array need to be set
+ */
+function createSampHsetCommand(qbObj, sampObj, aspectObj) {
+  const props = [];
+  if (qbObj[sampFields.MSG_BODY]) {
+    props.push(sampFields.MSG_BODY, qbObj[sampFields.MSG_BODY]);
+  }
+
+  if (qbObj[sampFields.MSG_CODE]) {
+    props.push(sampFields.MSG_CODE, qbObj[sampFields.MSG_CODE]);
+  }
+
+  if (qbObj[sampFields.NAME]) {
+    props.push(sampFields.NAME, qbObj[sampFields.NAME]);
+  }
+
+  /* calculate status value, update status, previous status and changed at
+  fields in 2 cases: a) sampObj not present b) sample obj present and
+   previous status != current status. Always update status. */
+  let value;
+  if (qbObj[sampFields.VALUE]) {
+    value = qbObj[sampFields.VALUE];
+  } else if (!sampObj) {
+    value = '';
+  }
+
+  if (value) {
+    const status = sampleUtils.computeStatus(aspectObj, value);
+
+    if (!sampObj || (sampObj && (sampObj[sampFields.PRVS_STATUS] !== status))) {
+      const prevStatus = sampObj ? sampObj[sampFields.STATUS] :
+                            dbConstants.statuses.Invalid;
+      props.push(sampFields.PRVS_STATUS, prevStatus);
+      props.push(sampFields.STS_CHNGED_AT, new Date().toString());
+      props.push(sampFields.STATUS, status);
+    }
+  }
```

Comment:
Any reason why you are using the props array? You can update the qbObj with the new fields and pass it to the hmset command.